containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/org/hawkular/openshift/namespace/NamespaceOverrideMapper.java

```diff
@@ -0,0 +1,279 @@
+import java.util.Map;
+import java.util.Set;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+
+import org.jboss.logging.Logger;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.JdkSSLOptions;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.SSLOptions;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.exceptions.InvalidQueryException;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ *
+ */
+public class NamespaceOverrideMapper {
+
+    private static final Logger log = Logger.getLogger(NamespaceOverrideMapper.class);
+
+    private Session session;
+
+    private final String token;
+
+    public Map<String,String> overrides;
+
+    public NamespaceOverrideMapper(String token) {
+        log.debug("Creating NamespaceOverrideMapper");
+
+        this.token = token;
+
+        boolean connected = false;
+        do {
+            try {
+                session = createSession();
+                connected = true;
+            } catch (Exception e) {
+                log.error("Could not connect to Cassandra. This could mean Cassandra is not yet up and running. Will try again. Error message: " + e.getLocalizedMessage());
+                try {
+                    Thread.sleep(2000);
+                } catch (InterruptedException e1) {
+                    e1.printStackTrace();
+                }
+            }
+        } while (!connected);
+        log.debug("Connection to Cassandra establised");
+
+        // ensure that the openshift_metrics keyspace exists
+        ensureKeyspace("openshift_metrics");
+
+        // if we already have a mappingTable stored in OpenShift, then use it.
+        // Otherwise we need to generate the overwrite table.
+        if (hasMappingTable()) {
+            log.debug("Mapping table already exists in Cassandra. Using this table");
+            this.overrides = getMappingTable();
+        } else {
+            log.debug("Mapping table does not exist in Cassandra. Creating Mapping");
+            // get the current list of tenants in Hawkular Metrics
+            Set<String> tenants = getTenants();
+
+            // map the current tenants to the current projects
+            Map<String, String> namespaceMapping = getNamespaceMapping(tenants);
+            log.debug("Namespace mapping: " + namespaceMapping);
+
+            this.overrides = namespaceMapping;
+
+            // write this mapping table back to Cassandra. We don't want to ever recalculate this table
+            log.debug("Writing the mapping table to Cassandra");
+            writeMappingTable(this.overrides);
+            log.debug("Mapping table stored in Cassandra");
+        }
+    }
+
+    private Session createSession() {
```

Comment:
I am not sure if it is possible, but it would be better to reuse the same session that is used elsewhere in hawkular-metrics.